core/src/main/java/jenkins/model/queue/AsynchronousExecution.java

```diff
@@ -106,7 +106,11 @@ public synchronized final Executor getExecutor() {
         return executor;
     }
 
-    @Restricted(NoExternalUse.class)
+    /**
+     * Use {@link #setExecutorWithoutCompleting(Executor)} and {@link #maybeComplete()}
+     * instead. For this method, locking behaviour is problematic.
+     */
+    @Restricted(NoExternalUse.class) @Deprecated()
     public synchronized final void setExecutor(@Nonnull Executor executor) {
         assert this.executor==null;
```

Comment:
Replace code by invocation of existing methods?

Reply:
Good idea, done.